File: core/src/test/java/xtdb/util/RowCounterTest.java
package xtdb.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class RowCounterTest {

    @Test
    void testRowCounter() {
        var rc = new RowCounter(24);

        assertEquals(24, rc.getChunkIdx());

        assertEquals(0, rc.getChunkRowCount());

        rc.addRows(15);
        assertEquals(15, rc.getChunkRowCount());

        rc.addRows(15);
        assertEquals(30, rc.getChunkRowCount());
    }
}

File: core/src/test/kotlin/xtdb/util/RowCounterTest.kt
package xtdb.util

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class RowCounterTest {
    @Test
    fun testRowCounter() {
        val rc = RowCounter(24)

        assertEquals(24, rc.blockIdx)
        assertEquals(0, rc.blockRowCount)

        rc.addRows(15)
        assertEquals(15, rc.blockRowCount)

        rc.addRows(15)
        assertEquals(30, rc.blockRowCount)
    }
}
